packages/next/src/build/webpack/loaders/next-app-loader.ts

```diff
@@ -93,10 +106,26 @@ async function createTreeCodeFromPath({
         })
       )
 
+      const layout = filePaths.find(
+        ([type, path]) => type === 'layout' && !!path
+      )?.[1]
+
+      // Collect metadata for the layout
+      if (layout) {
+        metadataCode += `{
+          type: 'layout',
+          mod: () => import(/* webpackMode: "eager" */ ${JSON.stringify(
+            layout
+          )}),
```

Comment:
Does this work if you "use client" in your `layout.js` or `page.js` file?

Reply:
Not yet, I'm making a SWC transform plugin to make sure that works.